core/services/relay/evm/mercury/transmitter.go

```diff
@@ -166,32 +206,37 @@ func (mt *mercuryTransmitter) runloop() {
 
 		b.Reset()
 		if res.Error == "" {
+			mt.transmitSuccessCount.Inc()
 			mt.lggr.Debugw("Transmit report success", "req", t.Req, "response", res, "reportCtx", t.ReportCtx)
 		} else {
 			// We don't need to retry here because the mercury server
 			// has confirmed it received the report. We only need to retry
 			// on networking/unknown errors
 			switch res.Code {
 			case DuplicateReport:
+				mt.transmitSuccessCount.Inc()
+				mt.transmitDuplicateCount.Inc()
 				mt.lggr.Debugw("Transmit report succeeded; duplicate report", "code", res.Code)
 			default:
 				elems := map[string]interface{}{}
 				var validFrom int64
 				var currentBlock int64
+				var unpackErr error
 				if err = PayloadTypes.UnpackIntoMap(elems, t.Req.Payload); err != nil {
-					mt.lggr.Errorw("failed to unpack report", "err", err)
+					unpackErr = err
 				} else {
 					report := elems["report"].([]byte)
 					validFrom, err = (&reportcodec.EVMReportCodec{}).ValidFromBlockNumFromReport(report)
 					if err != nil {
-						mt.lggr.Errorw("failed to unpack report", "err", err)
+						unpackErr = err
 					}
 					currentBlock, err = (&reportcodec.EVMReportCodec{}).CurrentBlockNumFromReport(report)
 					if err != nil {
-						mt.lggr.Errorw("failed to unpack report", "err", err)
+						unpackErr = errors.Join(unpackErr, err)
 					}
 				}
-				mt.lggr.Errorw("Transmit report failed; mercury server returned error", "validFromBlock", validFrom, "currentBlock", currentBlock, "req", t.Req, "response", res, "reportCtx", t.ReportCtx, "err", res.Error, "code", res.Code)
+				transmitServerErrorCount.WithLabelValues(mt.feedIDHex, fmt.Sprintf("%d", res.Code)).Inc()
```

Comment:
Shouldn't `transmitServerErrorCount` be part of `mercuryTransmitter` struct like the others?

Reply:
It can't be because it needs a dynamic label based on code